npm install fails because contextify cannot be installed on node 4.2.1 and OSX El Capitan 10.11

Here is a link to the to the existing issue about contextify: https://github.com/brianmcd/contextify/issues/188

Here is my npm install log:
```
npm install
npm WARN engine jest-cli@0.4.19: wanted: {"node":"0.8.x || 0.10.x"} (current: {"node":"4.2.1","npm":"2.14.7"})
 
> contextify@0.1.14 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/jest-cli/node_modules/jsdom/node_modules/contextify
> node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
In file included from ../src/contextify.cc:3:
../node_modules/nan/nan.h:261:25: error: redefinition of '_NanEnsureLocal'
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                        ^
../node_modules/nan/nan.h:256:25: note: previous definition is here
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                        ^
../node_modules/nan/nan.h:661:13: error: no member named 'smalloc' in namespace 'node'
    , node::smalloc::FreeCallback callback
      ~~~~~~^
../node_modules/nan/nan.h:672:12: error: no matching function for call to 'New'
    return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
           ^~~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:31:40: note: candidate function not viable: no known conversion from 'uint32_t'
      (aka 'unsigned int') to 'enum encoding' for 3rd argument
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:43:40: note: candidate function not viable: 2nd argument ('const char *') would
      lose const qualifier
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:28:40: note: candidate function not viable: requires 2 arguments, but 3 were
      provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:36:40: note: candidate function not viable: requires 5 arguments, but 3 were
      provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
In file included from ../src/contextify.cc:3:
../node_modules/nan/nan.h:676:12: error: no viable conversion from 'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object>'
    return node::Buffer::New(v8::Isolate::GetCurrent(), size);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:210:7: note: candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'v8::MaybeLocal<v8::Object>' to 'const v8::Local<v8::Object> &' for 1st argument
class Local {
      ^
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:210:7: note: candidate constructor (the implicit move constructor) not viable: no known
      conversion from 'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object> &&' for 1st argument
class Local {
      ^
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:214:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:326:13: note: candidate template ignored: could not match 'S *' against
      'v8::MaybeLocal<v8::Object>'
  V8_INLINE Local(S* that)
            ^
In file included from ../src/contextify.cc:3:
../node_modules/nan/nan.h:683:26: error: no member named 'Use' in namespace 'node::Buffer'
    return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
           ~~~~~~~~~~~~~~^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Primitive *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:414:12: note: in instantiation of function template specialization 'v8::Local<v8::Primitive>::Local<v8::Value>'
      requested here
    return NanEscapeScope(NanNew(v8::Undefined(v8::Isolate::GetCurrent())));
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Boolean *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:424:12: note: in instantiation of function template specialization 'v8::Local<v8::Boolean>::Local<v8::Value>'
      requested here
    return NanEscapeScope(NanNew(v8::True(v8::Isolate::GetCurrent())));
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Function *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1514:12: note: in instantiation of function template specialization 'v8::Local<v8::Function>::Local<v8::Value>'
      requested here
    return NanEscapeScope(NanNew(handle)->Get(kCallbackIndex)
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Object *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1632:12: note: in instantiation of function template specialization 'v8::Local<v8::Object>::Local<v8::Value>'
      requested here
    return NanEscapeScope(handle->Get(NanNew(key)).As<v8::Object>());
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Value *volatile' from incompatible type 'v8::Context *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:267:10: note: in instantiation of function template specialization 'v8::Local<v8::Value>::Local<v8::Context>'
      requested here
  return NanNew(val);
         ^
../src/contextify.cc:159:16: note: in instantiation of function template specialization '_NanEnsureLocal<v8::Local<v8::Context> >'
      requested here
        return NanEscapeScope(NanNew<Context>(
               ^
../node_modules/nan/nan.h:398:43: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                                          ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Context *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../src/contextify.cc:159:16: note: in instantiation of function template specialization 'v8::Local<v8::Context>::Local<v8::Value>'
      requested here
        return NanEscapeScope(NanNew<Context>(
               ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Integer *volatile' from incompatible type 'v8::Value *'
  TYPE_CHECK(T, S);
  ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../src/contextify.cc:211:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Integer>::Set<v8::Value>'
      requested here
            NanReturnValue(NanNew<Integer>(None));
            ^
../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
# define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
                                                            ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Boolean *volatile' from incompatible type 'v8::Value *'
  TYPE_CHECK(T, S);
  ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../src/contextify.cc:225:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Boolean>::Set<v8::Value>'
      requested here
            NanReturnValue(NanNew<Boolean>(false));
            ^
../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
# define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
                                                            ^
In file included from ../src/contextify.cc:1:
In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Array *volatile' from incompatible type 'v8::Value *'
  TYPE_CHECK(T, S);
  ^~~~~~~~~~~~~~~~
/Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../src/contextify.cc:236:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Array>::Set<v8::Value>'
      requested here
            NanReturnValue(blank);
            ^
../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
# define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
                                                            ^
14 errors generated.
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/eugene/htdocs/nodejs_projects/flux/node_modules/jest-cli/node_modules/jsdom/node_modules/contextify
gyp ERR! node -v v4.2.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
\
> fsevents@1.0.2 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/babel/node_modules/chokidar/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/eugene/htdocs/nodejs_projects/flux/node_modules/babel/node_modules/chokidar/node_modules/fsevents/lib/binding/Release/node-v46-darwin-x64/fse.node" is installed via remote

> fsevents@1.0.2 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/eugene/htdocs/nodejs_projects/flux/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents/lib/binding/Release/node-v46-darwin-x64/fse.node" is installed via remote
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.2.1
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE

npm ERR! contextify@0.1.14 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.14 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/eugene/htdocs/nodejs_projects/flux/npm-debug.log
```

Why are you using 4.x when jest-cli 0.4.x requires 0.8 - 0.10?


Sent from my iPhone

> On Oct 22, 2015, at 11:00 AM, epavlenko <notifications@github.com> wrote:
> 
> Here is a link to the to the existing issue about contextify: brianmcd/contextify#188
> 
> Here is my npm install log:
> 
> npm install
> npm WARN engine jest-cli@0.4.19: wanted: {"node":"0.8.x || 0.10.x"} (current: {"node":"4.2.1","npm":"2.14.7"})
> 
> > contextify@0.1.14 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/jest-cli/node_modules/jsdom/node_modules/contextify
> > node-gyp rebuild
> 
>   CXX(target) Release/obj.target/contextify/src/contextify.o
> In file included from ../src/contextify.cc:3:
> ../node_modules/nan/nan.h:261:25: error: redefinition of '_NanEnsureLocal'
> NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
>                         ^
> ../node_modules/nan/nan.h:256:25: note: previous definition is here
> NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
>                         ^
> ../node_modules/nan/nan.h:661:13: error: no member named 'smalloc' in namespace 'node'
>     , node::smalloc::FreeCallback callback
>       ~~~~~~^
> ../node_modules/nan/nan.h:672:12: error: no matching function for call to 'New'
>     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
>            ^~~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:31:40: note: candidate function not viable: no known conversion from 'uint32_t'
>       (aka 'unsigned int') to 'enum encoding' for 3rd argument
> NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
>                                        ^
> /Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:43:40: note: candidate function not viable: 2nd argument ('const char *') would
>       lose const qualifier
> NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
>                                        ^
> /Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:28:40: note: candidate function not viable: requires 2 arguments, but 3 were
>       provided
> NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
>                                        ^
> /Users/eugene/.node-gyp/4.2.1/include/node/node_buffer.h:36:40: note: candidate function not viable: requires 5 arguments, but 3 were
>       provided
> NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
>                                        ^
> In file included from ../src/contextify.cc:3:
> ../node_modules/nan/nan.h:676:12: error: no viable conversion from 'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object>'
>     return node::Buffer::New(v8::Isolate::GetCurrent(), size);
>            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:210:7: note: candidate constructor (the implicit copy constructor) not viable: no known
>       conversion from 'v8::MaybeLocal<v8::Object>' to 'const v8::Local<v8::Object> &' for 1st argument
> class Local {
>       ^
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:210:7: note: candidate constructor (the implicit move constructor) not viable: no known
>       conversion from 'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object> &&' for 1st argument
> class Local {
>       ^
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:214:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
>   V8_INLINE Local(Local<S> that)
>             ^
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:326:13: note: candidate template ignored: could not match 'S *' against
>       'v8::MaybeLocal<v8::Object>'
>   V8_INLINE Local(S* that)
>             ^
> In file included from ../src/contextify.cc:3:
> ../node_modules/nan/nan.h:683:26: error: no member named 'Use' in namespace 'node::Buffer'
>     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
>            ~~~~~~~~~~~~~~^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Primitive *volatile' from incompatible type 'v8::Value *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../node_modules/nan/nan.h:414:12: note: in instantiation of function template specialization 'v8::Local<v8::Primitive>::Local<v8::Value>'
>       requested here
>     return NanEscapeScope(NanNew(v8::Undefined(v8::Isolate::GetCurrent())));
>            ^
> ../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                              ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Boolean *volatile' from incompatible type 'v8::Value *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../node_modules/nan/nan.h:424:12: note: in instantiation of function template specialization 'v8::Local<v8::Boolean>::Local<v8::Value>'
>       requested here
>     return NanEscapeScope(NanNew(v8::True(v8::Isolate::GetCurrent())));
>            ^
> ../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                              ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Function *volatile' from incompatible type 'v8::Value *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../node_modules/nan/nan.h:1514:12: note: in instantiation of function template specialization 'v8::Local<v8::Function>::Local<v8::Value>'
>       requested here
>     return NanEscapeScope(NanNew(handle)->Get(kCallbackIndex)
>            ^
> ../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                              ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Object *volatile' from incompatible type 'v8::Value *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../node_modules/nan/nan.h:1632:12: note: in instantiation of function template specialization 'v8::Local<v8::Object>::Local<v8::Value>'
>       requested here
>     return NanEscapeScope(handle->Get(NanNew(key)).As<v8::Object>());
>            ^
> ../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                              ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Value *volatile' from incompatible type 'v8::Context *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../node_modules/nan/nan.h:267:10: note: in instantiation of function template specialization 'v8::Local<v8::Value>::Local<v8::Context>'
>       requested here
>   return NanNew(val);
>          ^
> ../src/contextify.cc:159:16: note: in instantiation of function template specialization '_NanEnsureLocal<v8::Local<v8::Context> >'
>       requested here
>         return NanEscapeScope(NanNew<Context>(
>                ^
> ../node_modules/nan/nan.h:398:43: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                                           ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:221:5: error: assigning to 'v8::Context *volatile' from incompatible type 'v8::Value *'
>     TYPE_CHECK(T, S);
>     ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../src/contextify.cc:159:16: note: in instantiation of function template specialization 'v8::Local<v8::Context>::Local<v8::Value>'
>       requested here
>         return NanEscapeScope(NanNew<Context>(
>                ^
> ../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
> # define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
>                              ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Integer *volatile' from incompatible type 'v8::Value *'
>   TYPE_CHECK(T, S);
>   ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../src/contextify.cc:211:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Integer>::Set<v8::Value>'
>       requested here
>             NanReturnValue(NanNew<Integer>(None));
>             ^
> ../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
> # define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
>                                                             ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Boolean *volatile' from incompatible type 'v8::Value *'
>   TYPE_CHECK(T, S);
>   ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../src/contextify.cc:225:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Boolean>::Set<v8::Value>'
>       requested here
>             NanReturnValue(NanNew<Boolean>(false));
>             ^
> ../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
> # define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
>                                                             ^
> In file included from ../src/contextify.cc:1:
> In file included from /Users/eugene/.node-gyp/4.2.1/include/node/node.h:42:
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:7380:3: error: assigning to 'v8::Array *volatile' from incompatible type 'v8::Value *'
>   TYPE_CHECK(T, S);
>   ^~~~~~~~~~~~~~~~
> /Users/eugene/.node-gyp/4.2.1/include/node/v8.h:180:37: note: expanded from macro 'TYPE_CHECK'
>     *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
>                                     ^ ~~~~~~~~~~~~~~~~~~
> ../src/contextify.cc:236:13: note: in instantiation of function template specialization 'v8::ReturnValue<v8::Array>::Set<v8::Value>'
>       requested here
>             NanReturnValue(blank);
>             ^
> ../node_modules/nan/nan.h:401:61: note: expanded from macro 'NanReturnValue'
> # define NanReturnValue(value) return args.GetReturnValue().Set(_NanEnsureLocal(value))
>                                                             ^
> 14 errors generated.
> make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
> gyp ERR! build error 
> gyp ERR! stack Error: `make` failed with exit code: 2
> gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
> gyp ERR! stack     at emitTwo (events.js:87:13)
> gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
> gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
> gyp ERR! System Darwin 15.0.0
> gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
> gyp ERR! cwd /Users/eugene/htdocs/nodejs_projects/flux/node_modules/jest-cli/node_modules/jsdom/node_modules/contextify
> gyp ERR! node -v v4.2.1
> gyp ERR! node-gyp -v v3.0.3
> gyp ERR! not ok 
> \
> > fsevents@1.0.2 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/babel/node_modules/chokidar/node_modules/fsevents
> > node-pre-gyp install --fallback-to-build
> 
> [fsevents] Success: "/Users/eugene/htdocs/nodejs_projects/flux/node_modules/babel/node_modules/chokidar/node_modules/fsevents/lib/binding/Release/node-v46-darwin-x64/fse.node" is installed via remote
> 
> > fsevents@1.0.2 install /Users/eugene/htdocs/nodejs_projects/flux/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents
> > node-pre-gyp install --fallback-to-build
> 
> [fsevents] Success: "/Users/eugene/htdocs/nodejs_projects/flux/node_modules/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/fsevents/lib/binding/Release/node-v46-darwin-x64/fse.node" is installed via remote
> npm ERR! Darwin 15.0.0
> npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
> npm ERR! node v4.2.1
> npm ERR! npm  v2.14.7
> npm ERR! code ELIFECYCLE
> 
> npm ERR! contextify@0.1.14 install: `node-gyp rebuild`
> npm ERR! Exit status 1
> npm ERR! 
> npm ERR! Failed at the contextify@0.1.14 install script 'node-gyp rebuild'.
> npm ERR! This is most likely a problem with the contextify package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node-gyp rebuild
> npm ERR! You can get their info via:
> npm ERR!     npm owner ls contextify
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! Please include the following file with any support request:
> npm ERR!     /Users/eugene/htdocs/nodejs_projects/flux/npm-debug.log
> —
> Reply to this email directly or view it on GitHub.
>

Failing to build also on Debian here, 

```
$ nvm use v4.1.1
Now using node v4.1.1 (npm v3.3.5)
top@Bkkdaily:~/flux/examples/flux-chat$ npm install
npm WARN engine jest-cli@0.4.19: wanted: {"node":"0.8.x || 0.10.x"} (current: {"node":"4.1.1","npm":"3.3.5"})
npm WARN install Couldn't install optional dependency: Unsupported
npm WARN prefer global coffee-script@1.10.0 should be installed with -g
npm WARN prefer global cover@0.2.9 should be installed with -g

> contextify@0.1.14 install /home/top/flux/examples/flux-chat/node_modules/contextify
> node-gyp rebuild

make: Entering directory `/home/top/flux/examples/flux-chat/node_modules/contextify/build'
  CXX(target) Release/obj.target/contextify/src/contextify.o
In file included from ../src/contextify.cc:3:0:
../../nan/nan.h:261:25: error: redefinition of ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Local<T>)’
../../nan/nan.h:256:25: error: ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Handle<T>)’ previously declared here
../../nan/nan.h:661:13: error: ‘node::smalloc’ has not been declared
../../nan/nan.h:661:35: error: expected ‘,’ or ‘...’ before ‘callback’
../../nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(char*, size_t, int)’:
../../nan/nan.h:665:50: error: ‘callback’ was not declared in this scope
../../nan/nan.h:665:60: error: ‘hint’ was not declared in this scope
../../nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(const char*, uint32_t)’:
../../nan/nan.h:672:67: error: call of overloaded ‘New(v8::Isolate*, const char*&, uint32_t&)’ is ambiguous
../../nan/nan.h:672:67: note: candidates are:
In file included from ../../nan/nan.h:25:0,
                 from ../src/contextify.cc:3:
/home/top/.node-gyp/4.1.1/include/node/node_buffer.h:31:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) <near match>
/home/top/.node-gyp/4.1.1/include/node/node_buffer.h:31:40: note:   no known conversion for argument 3 from ‘uint32_t {aka unsigned int}’ to ‘node::encoding’
/home/top/.node-gyp/4.1.1/include/node/node_buffer.h:43:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, char*, size_t) <near match>
/home/top/.node-gyp/4.1.1/include/node/node_buffer.h:43:40: note:   no known conversion for argument 2 from ‘const char*’ to ‘char*’
In file included from ../src/contextify.cc:3:0:
../../nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(uint32_t)’:
../../nan/nan.h:676:61: error: could not convert ‘node::Buffer::New(v8::Isolate::GetCurrent(), ((size_t)size))’ from ‘v8::MaybeLocal<v8::Object>’ to ‘v8::Local<v8::Object>’
../../nan/nan.h: In function ‘v8::Local<v8::Object> NanBufferUse(char*, uint32_t)’:
../../nan/nan.h:683:12: error: ‘Use’ is not a member of ‘node::Buffer’
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
make: Leaving directory `/home/top/flux/examples/flux-chat/node_modules/contextify/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.2.0-4-amd64
gyp ERR! command "/home/top/.nvm/versions/node/v4.1.1/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/top/flux/examples/flux-chat/node_modules/contextify
gyp ERR! node -v v4.1.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
flux-chat@0.0.2 /home/top/flux/examples/flux-chat
├── bluebird@2.9.34  extraneous
├─┬ browserify@6.3.4
│ ├─┬ browser-pack@3.2.0
│ │ ├─┬ combine-source-map@0.3.0
│ │ │ ├─┬ inline-source-map@0.3.1
│ │ │ │ └─┬ source-map@0.3.0
│ │ │ │   └── amdefine@1.0.0 
│ │ │ └─┬ source-map@0.1.43
│ │ │   └── amdefine@1.0.0 
│ │ └─┬ through2@0.5.1
│ │   └─┬ readable-stream@1.0.33
│ │     └── core-util-is@1.0.1 
│ ├─┬ browser-resolve@1.10.1
│ │ └── resolve@1.1.6 
│ ├─┬ deps-sort@1.3.9
│ │ └─┬ JSONStream@1.0.6
│ │   └── through@2.3.8 
│ ├─┬ glob@4.5.3
│ │ ├─┬ inflight@1.0.4 
│ │ │ └── wrappy@1.0.1 
│ │ ├─┬ minimatch@2.0.10 
│ │ │ └─┬ brace-expansion@1.1.1 
│ │ │   ├── balanced-match@0.2.1 
│ │ │   └── concat-map@0.0.1 
│ │ └─┬ once@1.3.2 
│ │   └── wrappy@1.0.1 
│ ├── inherits@2.0.1 
│ ├─┬ insert-module-globals@6.6.3
│ │ ├─┬ combine-source-map@0.6.1
│ │ │ └─┬ source-map@0.4.4 
│ │ │   └── amdefine@1.0.0 
│ │ ├─┬ JSONStream@1.0.6
│ │ │ └── through@2.3.8 
│ │ └── xtend@4.0.0 
│ ├── isarray@0.0.1 
│ ├─┬ JSONStream@0.8.4
│ │ └── through@2.3.8 
│ ├─┬ module-deps@3.9.1
│ │ ├─┬ detective@4.2.0
│ │ │ └─┬ escodegen@1.7.0 
│ │ │   ├── esprima@1.2.5 
│ │ │   ├── estraverse@1.9.3 
│ │ │   ├── esutils@2.0.2 
│ │ │   ├─┬ optionator@0.5.0 
│ │ │   │ ├── deep-is@0.1.3 
│ │ │   │ ├── fast-levenshtein@1.0.7 
│ │ │   │ ├── levn@0.2.5 
│ │ │   │ ├── prelude-ls@1.1.2 
│ │ │   │ ├── type-check@0.3.1 
│ │ │   │ └── wordwrap@0.0.3 
│ │ │   └─┬ source-map@0.2.0 
│ │ │     └── amdefine@1.0.0 
│ │ ├─┬ JSONStream@1.0.6
│ │ │ └── through@2.3.8 
│ │ ├── resolve@1.1.6 
│ │ ├─┬ stream-combiner2@1.0.2
│ │ │ └─┬ through2@0.5.1
│ │ │   └─┬ readable-stream@1.0.33
│ │ │     └── core-util-is@1.0.1 
│ │ └── xtend@4.0.0 
│ ├─┬ readable-stream@1.1.13 
│ │ └── core-util-is@1.0.1 
│ ├── string_decoder@0.10.31 
│ ├─┬ through2@1.1.1
│ │ └── xtend@4.0.0 
│ └─┬ umd@2.1.0
│   ├─┬ ruglify@1.0.0
│   │ └─┬ uglify-js@2.2.5
│   │   ├─┬ optimist@0.3.7
│   │   │ └── wordwrap@0.0.3 
│   │   └─┬ source-map@0.1.43
│   │     └── amdefine@1.0.0 
│   └── through@2.3.8 
├── coffee-script@1.10.0  extraneous
├── cover@0.2.9  extraneous
├── cssstyle@0.2.30  extraneous
├── diff@1.0.8  extraneous
├─┬ envify@3.4.0
│ ├─┬ jstransform@10.1.0
│ │ └─┬ source-map@0.1.31
│ │   └── amdefine@1.0.0 
│ └── through@2.3.8 
├── graceful-fs@2.0.3  extraneous
├── harmonize@1.4.2  extraneous
├── htmlparser2@3.8.3  extraneous
├── istanbul@0.3.22  extraneous
├── jasmine-only@0.1.0  extraneous
├── jasmine-pit@2.0.2  extraneous
├── lodash.template@3.6.2  extraneous
├── node-haste@1.2.8  extraneous
├── node-worker-pool@2.4.5  extraneous
├── nwmatcher@1.3.6  extraneous
├─┬ reactify@0.15.2
│ ├─┬ jstransform@6.3.2
│ │ └─┬ source-map@0.1.31
│ │   └── amdefine@1.0.0 
│ ├─┬ react-tools@0.12.2
│ │ ├─┬ commoner@0.10.3
│ │ │ ├─┬ glob@4.2.2
│ │ │ │ ├─┬ inflight@1.0.4 
│ │ │ │ │ └── wrappy@1.0.1 
│ │ │ │ ├── inherits@2.0.1 
│ │ │ │ └─┬ once@1.3.2 
│ │ │ │   └── wrappy@1.0.1 
│ │ │ └─┬ mkdirp@0.5.1 
│ │ │   └── minimist@0.0.8 
│ │ └─┬ jstransform@8.2.0
│ │   └─┬ source-map@0.1.31
│ │     └── amdefine@1.0.0 
│ └── through@2.3.8 
├── request@2.65.0  extraneous
├── resolve@0.6.3  extraneous
├── through@2.3.8  extraneous
├─┬ uglify-js@2.4.24
│ └─┬ source-map@0.1.34
│   └── amdefine@1.0.0 
├─┬ watchify@2.6.2
│ ├─┬ browserify@9.0.8
│ │ ├─┬ browser-pack@4.0.4
│ │ │ ├─┬ combine-source-map@0.3.0
│ │ │ │ ├─┬ inline-source-map@0.3.1
│ │ │ │ │ └─┬ source-map@0.3.0
│ │ │ │ │   └── amdefine@1.0.0 
│ │ │ │ └─┬ source-map@0.1.43
│ │ │ │   └── amdefine@1.0.0 
│ │ │ ├─┬ JSONStream@1.0.6
│ │ │ │ └── through@2.3.8 
│ │ │ └─┬ through2@0.5.1
│ │ │   └─┬ readable-stream@1.0.33
│ │ │     └── core-util-is@1.0.1 
│ │ ├─┬ deps-sort@1.3.9
│ │ │ └─┬ JSONStream@1.0.6
│ │ │   └── through@2.3.8 
│ │ ├─┬ glob@4.5.3
│ │ │ ├─┬ inflight@1.0.4 
│ │ │ │ └── wrappy@1.0.1 
│ │ │ ├─┬ minimatch@2.0.10 
│ │ │ │ └─┬ brace-expansion@1.1.1 
│ │ │ │   ├── balanced-match@0.2.1 
│ │ │ │   └── concat-map@0.0.1 
│ │ │ └─┬ once@1.3.2 
│ │ │   └── wrappy@1.0.1 
│ │ ├── inherits@2.0.1 
│ │ ├─┬ insert-module-globals@6.6.3
│ │ │ ├─┬ combine-source-map@0.6.1
│ │ │ │ └─┬ source-map@0.4.4 
│ │ │ │   └── amdefine@1.0.0 
│ │ │ ├─┬ JSONStream@1.0.6
│ │ │ │ └── through@2.3.8 
│ │ │ └── xtend@4.0.0 
│ │ ├── isarray@0.0.1 
│ │ ├─┬ JSONStream@0.10.0
│ │ │ └── through@2.3.8 
│ │ ├─┬ module-deps@3.9.1
│ │ │ ├─┬ detective@4.2.0
│ │ │ │ └─┬ escodegen@1.7.0 
│ │ │ │   ├── esprima@1.2.5 
│ │ │ │   ├── estraverse@1.9.3 
│ │ │ │   ├── esutils@2.0.2 
│ │ │ │   ├─┬ optionator@0.5.0 
│ │ │ │   │ ├── deep-is@0.1.3 
│ │ │ │   │ ├── fast-levenshtein@1.0.7 
│ │ │ │   │ ├── levn@0.2.5 
│ │ │ │   │ ├── prelude-ls@1.1.2 
│ │ │ │   │ ├── type-check@0.3.1 
│ │ │ │   │ └── wordwrap@0.0.3 
│ │ │ │   └─┬ source-map@0.2.0 
│ │ │ │     └── amdefine@1.0.0 
│ │ │ ├─┬ JSONStream@1.0.6
│ │ │ │ └── through@2.3.8 
│ │ │ ├─┬ stream-combiner2@1.0.2
│ │ │ │ └─┬ through2@0.5.1
│ │ │ │   └─┬ readable-stream@1.0.33
│ │ │ │     └── core-util-is@1.0.1 
│ │ │ └── xtend@4.0.0 
│ │ ├─┬ readable-stream@1.1.13 
│ │ │ └── core-util-is@1.0.1 
│ │ ├── resolve@1.1.6 
│ │ ├── string_decoder@0.10.31 
│ │ └─┬ through2@1.1.1
│ │   └── xtend@4.0.0 
│ ├─┬ chokidar@0.12.6
│ │ └─┬ readdirp@1.3.0
│ │   ├── graceful-fs@2.0.3 
│ │   └─┬ readable-stream@1.0.33
│ │     ├── core-util-is@1.0.1 
│ │     ├── inherits@2.0.1 
│ │     ├── isarray@0.0.1 
│ │     └── string_decoder@0.10.31 
│ ├─┬ through2@0.5.1
│ │ └─┬ readable-stream@1.0.33
│ │   ├── core-util-is@1.0.1 
│ │   ├── inherits@2.0.1 
│ │   ├── isarray@0.0.1 
│ │   └── string_decoder@0.10.31 
│ └── xtend@4.0.0 
└── xmlhttprequest@1.8.0  extraneous

npm WARN EPACKAGEJSON flux-chat@0.0.2 No license field.
npm ERR! Linux 3.2.0-4-amd64
npm ERR! argv "/home/top/.nvm/versions/node/v4.1.1/bin/node" "/usr/bin/npm" "install"
npm ERR! node v4.1.1
npm ERR! npm  v3.3.5
npm ERR! code ELIFECYCLE

npm ERR! contextify@0.1.14 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.14 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/top/flux/examples/flux-chat/npm-debug.log
```

--- a/package.json
+++ b/package.json
@@ -43,11 +43,12 @@
     "babel-core": "^5.8.22",
     "babel-loader": "^5.3.2",
     "del": "^1.2.0",
+    "fbjs-scripts": "^0.2.2",
     "gulp": "^3.9.0",
     "gulp-babel": "^5.1.0",
     "gulp-flatten": "^0.1.1",
     "gulp-util": "^3.0.6",
-    "jest-cli": "^0.4.18",
+    "jest-cli": "^0.5.10",
     "object-assign": "^3.0.0",
     "run-sequence": "^1.1.0",
     "vinyl-source-stream": "^1.0.0",
@@ -56,7 +57,7 @@
   },
   "dependencies": {
     "fbemitter": "^2.0.0",
-    "fbjs": "0.1.0-alpha.7",
+    "fbjs": "^0.4.0",
     "immutable": "^3.7.4"
   },
   "jest": {

--- a/scripts/babel/default-options.js
+++ b/scripts/babel/default-options.js
@@ -11,8 +11,8 @@
 
 var assign = require('object-assign');
 
-var babelPluginDEV = require('fbjs/scripts/babel/dev-expression');
-var babelPluginModules = require('fbjs/scripts/babel/rewrite-modules');
+var babelPluginDEV = require('fbjs-scripts/babel/dev-expression');
+var babelPluginModules = require('fbjs-scripts/babel/rewrite-modules');
 
 var moduleMap = require('fbjs/module-map');
 

--- a/scripts/jest/environment.js
+++ b/scripts/jest/environment.js
@@ -1,1 +1,1 @@
-__DEV__ = true;
+global.__DEV__ = true;

--- a/src/__tests__/Dispatcher-test.js
+++ b/src/__tests__/Dispatcher-test.js
@@ -1,4 +1,4 @@
-/*
+/**
  * Copyright (c) 2014-2015, Facebook, Inc.
  * All rights reserved.
  *
@@ -10,9 +10,9 @@
 jest.dontMock('Dispatcher');
 jest.dontMock('invariant');
 
-describe('Dispatcher', () => {
+var Dispatcher = require('Dispatcher');
 
-  var Dispatcher = require('Dispatcher');
+describe('Dispatcher', () => {
   var dispatcher;
   var callbackA;
   var callbackB;
@@ -95,8 +95,7 @@ describe('Dispatcher', () => {
     expect(() => {
       dispatcher.dispatch(payload);
     }).toThrow(
-      'Invariant Violation: Dispatch.dispatch(...): Cannot dispatch in the ' +
-      'middle of a dispatch.'
+      'Dispatch.dispatch(...): Cannot dispatch in the middle of a dispatch.'
     );
 
     expect(callbackA.mock.calls.length).toBe(0);
@@ -108,8 +107,7 @@ describe('Dispatcher', () => {
     expect(() => {
       dispatcher.waitFor([tokenA]);
     }).toThrow(
-      'Invariant Violation: Dispatcher.waitFor(...): Must be invoked while ' +
-      'dispatching.'
+      'Dispatcher.waitFor(...): Must be invoked while dispatching.'
     );
 
     expect(callbackA.mock.calls.length).toBe(0);
@@ -126,8 +124,7 @@ describe('Dispatcher', () => {
     expect(() => {
       dispatcher.dispatch(payload);
     }).toThrow(
-      'Invariant Violation: Dispatcher.waitFor(...): `1337` does not map to ' +
-      'a registered callback.'
+      'Dispatcher.waitFor(...): `1337` does not map to a registered callback.'
     );
   });
 
@@ -141,8 +138,8 @@ describe('Dispatcher', () => {
     expect(() => {
       dispatcher.dispatch(payload);
     }).toThrow(
-      'Invariant Violation: Dispatcher.waitFor(...): Circular dependency ' +
-      'detected while waiting for `' + tokenA + '`.'
+      'Dispatcher.waitFor(...): Circular dependency detected while waiting ' +
+      'for `' + tokenA + '`.'
     );
 
     expect(callbackA.mock.calls.length).toBe(0);
@@ -162,8 +159,8 @@ describe('Dispatcher', () => {
     expect(() => {
       dispatcher.dispatch({});
     }).toThrow(
-      'Invariant Violation: Dispatcher.waitFor(...): Circular dependency ' +
-      'detected while waiting for `' + tokenA + '`.'
+      'Dispatcher.waitFor(...): Circular dependency detected while waiting ' +
+      'for `' + tokenA + '`.'
     );
 
     expect(callbackA.mock.calls.length).toBe(0);
@@ -215,5 +212,4 @@ describe('Dispatcher', () => {
 
     expect(callbackB.mock.calls.length).toBe(1);
   });
-
 });

--- a/src/__tests__/FluxStoreGroup-test.js
+++ b/src/__tests__/FluxStoreGroup-test.js
@@ -28,6 +28,7 @@ class SimpleStore extends FluxStore {
 describe('FluxStoreGroup', () => {
   var dispatcher;
   var storeA;
+  var storeB;
 
   beforeEach(() => {
     dispatcher = new Dispatcher();

--- a/src/stores/__tests__/FluxStore-test.js
+++ b/src/stores/__tests__/FluxStore-test.js
@@ -64,8 +64,7 @@ describe('FluxStore', () => {
     new IncompleteFluxStore(dispatcher);
     var incompleteStoreCallback = dispatcher.register.mock.calls[1][0];
     expect(() => incompleteStoreCallback({type: 'action-type'})).toThrow(
-      'Invariant Violation: IncompleteFluxStore has not overridden ' +
-      'FluxStore.__onDispatch(), which is required'
+      'IncompleteFluxStore has not overridden FluxStore.__onDispatch(), which is required'
     );
     expect(() => registeredCallback({type: 'action-type'})).not.toThrow();
   });
@@ -73,15 +72,13 @@ describe('FluxStore', () => {
   it('throws when __emitChange() is invoked outside of a dispatch', () => {
     var illegalFluxStore = new IllegalFluxStore(dispatcher);
     expect(() => illegalFluxStore.illegalEmit()).toThrow(
-      'Invariant Violation: IllegalFluxStore.__emitChange(): Must be invoked ' +
-      'while dispatching.'
+      'IllegalFluxStore.__emitChange(): Must be invoked while dispatching.'
     );
   });
 
   it('throws when hasChanged() is invoked outside of a dispatch', () => {
     expect(() => fluxStore.hasChanged()).toThrow(
-      'Invariant Violation: TestFluxStore.hasChanged(): Must be invoked ' +
-      'while dispatching.'
+      'TestFluxStore.hasChanged(): Must be invoked while dispatching.'
     );
   });
 

